File: netDxf/Entities/MLeaderLine.Fields.cs
// Copyright (c) netDxf contributors. Licensed under the MIT License.
using System;
using netDxf.Blocks;
using netDxf.Objects;
using netDxf.Tables;

namespace netDxf.Entities
{
    /// <summary>Editable stored MLeaderLine fields. Redundant values are retained independently; angles are radians and colors are DXF raw-color integers.</summary>
    public sealed partial class MLeaderLine : MLeaderData
    {
        private static readonly MLeaderField[] fields = new MLeaderField[]
        {
            new MLeaderField(91, typeof(int), 0, false, 2007),
            new MLeaderField(92, typeof(int), unchecked((int)0xC1000000), false, 2007),
        };
        internal override MLeaderField[] Fields { get { return fields; } }
        /// <summary>Gets or sets the stored group 91 value.</summary>
        public int Index { get { return this.Get<int>(91); } set { this.Set(91, value); } }
        /// <summary>Gets or sets the stored group 92 value.</summary>
        public int Color { get { return this.Get<int>(92); } set { this.Set(92, value); } }
    }
}

File: javascript/netDxf/Entities/MLeaderLine.Fields.js
// Copyright (c) netDxf contributors. MIT License; see package LICENSE.
import {MLeaderData,InstallMLeaderFields,MLeaderComponentTypes} from './MLeaderData.js';
import {InitializeMLeaderChildren,InstallMLeaderChildren} from './MLeaderContext.js';
import {Vector3} from '../Vector3.js';
export class MLeaderLine extends MLeaderData {
  constructor(){super();InitializeMLeaderChildren(this,'MLeaderLine');}
  get Color(){return this.StoredColor??-1056964608;} set Color(value){this.Set(92,value);}
}
InstallMLeaderFields(MLeaderLine,[
  ['Index',91,'int',0,false,2007],
  ['StoredColor',92,'int',null,false,2007],
]);
MLeaderComponentTypes.set('MLeaderLine',MLeaderLine);
InstallMLeaderChildren(MLeaderLine,'MLeaderLine');
